### PR TITLE
item_updater: add support MCU firmware image update

### PR DESCRIPTION
**Pull request by timlee66**
Tested:
In WebUI/Firmware page, Upload MCU F/W image then Active it.
Image state will become from Ready to Functional and firmware version will show on WebUI.

Signed-off-by: Tim Lee <timlee660101@gmail.com>

### CODE REVIEW

File: static/flash.cpp [hunk 1]
```diff
@@ -63,37 +66,54 @@ void HostActivation::flashWrite()
     log<level::DEBUG>("HostActivation::flashWrite");
     fs::path uploadDir(IMG_UPLOAD_DIR);
     // CPLD may update via this function
-    for (auto& bmcImage : phosphor::software::image::bmcImages)
+    if (fs::exists(uploadDir / versionId / BIOS_IMAGE))
     {
-        if(bmcImage.compare("image-bios")==0)
+        fs::copy_file(uploadDir / versionId / BIOS_IMAGE, "/tmp/image-bios",
+                          fs::copy_options::overwrite_existing);
```
Comment (minimada) [inline static/flash.cpp:75]: maybe just like above bmcImage use toPath / bmcImage, you can use toPath / BIOS_IMAGE

**Review by minimada — COMMENTED** [review]
I suggest this change should be a patch in openbmc, not merge to software manager.
Unless you have tested both the host and mcu image update progress is work well with this change, you should never put this to repository.

File: static/flash.cpp (hunk 1 above)
Comment (timlee66) [reply static/flash.cpp:75]: Done

**Comment by timlee66** [comment]
Done

File: static/flash.cpp [hunk 2]
```diff
@@ -105,7 +111,7 @@ void HostActivation::onStateChanges(sdbusplus::message::message& msg)
 
     // Read the msg and populate each variable
     msg.read(newStateID, newStateObjPath, newStateUnit, newStateResult);
-    if (newStateUnit == flashBiosServiceFile)
+    if ((newStateUnit == flashUpdateServiceFile))
```
Comment (minimada) [inline static/flash.cpp]: why need one more "()"?

Comment (timlee66) [reply static/flash.cpp]: Change it. Thanks.

**Pull request merged**